packages/Frontend/src/containers/AuthenticateContainer.jsx

```diff
@@ -23,47 +22,86 @@ class AuthenticateContainer extends Component {
     super(props);
     this.state = {
       email: '',
-      error: '',
+      error: false,
       loading: false,
       password: '',
     };
   }
 
+  componentDidMount = () => {
+    this.setState({ loading: true }, () => {
+      auth().getRedirectResult().then((result) => {
+        this.setState({ loading: false });
+        console.log('success', result);
+      }, (error) => {
+        this.setState({ loading: false });
+        // The provider's account email, can be used in case of
+        // auth/account-exists-with-different-credential to fetch the providers
+        // linked to the email:
+        console.error(error);
+        console.error(error.message);
+        console.error(error.code);
+        // In case of auth/account-exists-with-different-credential error,
+        // you can fetch the providers using this:
+        if (error.code === 'auth/account-exists-with-different-credential') {
+          auth().fetchProvidersForEmail(error.email).then((providers) => {
+            console.log('other providers', providers);
+          // The returned 'providers' is a list of the available providers
+          // linked to the email address. Please refer to the guide for a more
+          // complete explanation on how to recover from this error.
+          });
+        }
+      });
+    });
+  };
+
   updateFormByKey = key => ({ target }) => {
     this.setState({ [key]: target.value });
   };
 
-  authenticate = () => {
+  authenticate = (provider) => {
```

Comment:
I noticed similarities in how firebase handled authentication with Google and Facebook, so I put the similarities in another class method called authenticateWithProvider, this just calls them if it get's 'google' or 'facebook' passed to them as a string. This function runs whenever a user clicks on the new sign in with facebook and google buttons.